File: Python/binarySearch.py
def binary(lst1,item):
    l = len(lst1)
    lp,up = 0,l-1
    mp = (up+lp)//2
    
    if item not in lst1:
        print("Item not found in list")
        
    elif item >= lst1[mp]:
        for i in range(mp,l):
            if lst1[i] == item:
                print("Item",item,"found at index",i)
            
    elif item < lst1[mp]:
        for i in range(mp):
            if lst1[i] == item:
                print("Item",item,"found at index",i)
                
def traverse(l):
    print("Traversing the list: ")
    print(l[:])

def main():
  item = int(input("Enter item to search: "))
  l = map(int,input().split())
  binary(l,item)
  
if __name__ == '__main__':
  main()

File: C/binarySearch.c
// Binary Search
#include <stdio.h>
#include <stdbool.h>

void main() {
  int ar[] = {5,16,20,22,34,50};
  int len = sizeof(ar)/sizeof(int), key;  // key = 34, len = 6
  int first = 0, last = len-1; // first = 4, last = 5
  int mid = (first+last)/2;    // mid = 5
  bool found = false;
  
  scanf("%d",&key);
  
  if (ar[mid] == key)
  {
    printf("Found at index %d",mid);
    found = true;
  }
  else
  {
    while (first < last)
    {
      if (ar[mid] > key)
        last = mid - 1;
      else if (ar[mid] < key)
        first = mid + 1;
      else
      {
        found = true;
        printf("found at index %d",mid);
        break;
      }
      mid = (first + last)/2;
    }
  }
  
  if (found == false)
    printf("Did not find");
}
